app/src/main/java/com/spotlight/spotlightapp/task/views/DailyIntentListFragment.kt

```diff
@@ -0,0 +1,99 @@
+package com.spotlight.spotlightapp.task.views
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.spotlight.spotlightapp.R
+import com.spotlight.spotlightapp.data.task.Task
+import com.spotlight.spotlightapp.databinding.FragmentDailyIntentListBinding
+import com.spotlight.spotlightapp.task.TaskPageRouter
+import com.spotlight.spotlightapp.task.adapters.DailyIntentListAdapter
+import com.spotlight.spotlightapp.task.viewmodels.DailyIntentListViewModel
+import com.spotlight.spotlightapp.utilities.viewmodelutils.BaseViewModel
+import com.spotlight.spotlightapp.utilities.viewmodelutils.ViewModelErrorListener
+import com.spotlight.spotlightapp.utilities.viewmodelutils.observeErrors
+import com.spotlight.spotlightapp.utilities.viewutils.MarginItemDecoration
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class DailyIntentListFragment(private val taskPageRouter: TaskPageRouter)
+    : Fragment(R.layout.fragment_daily_intent_list), ViewModelErrorListener,
+    DailyIntentListAdapter.Callback {
+    private val viewModel: DailyIntentListViewModel by viewModels()
+    private lateinit var viewBinding: FragmentDailyIntentListBinding
+    private lateinit var dailyIntentListAdapter: DailyIntentListAdapter
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewBinding = FragmentDailyIntentListBinding.bind(view)
+        configureViews()
+        observeViewModel()
+        observeErrors()
+    }
+
+    override val baseViewModel: BaseViewModel
+        get() = viewModel
+
+    override val snackbarLayout: View
+        get() = viewBinding.root
+
+    override fun onTaskSelected(view: View, task: Task) {
+        taskPageRouter.openTaskPage(view, task, false)
+    }
+
+    private fun configureViews() {
+        configureTasksRecyclerView()
+        configureAddButton()
+    }
+
+    private fun configureTasksRecyclerView() {
+        dailyIntentListAdapter = DailyIntentListAdapter(this)
```

Comment:
This should be initialized from the start instead to optimize configuration of `tasksRecyclerView`.